internal/controller/storageclient_controller.go

```diff
@@ -779,27 +661,33 @@ func removeStorageClaimAsOwner(obj client.Object) {
 	}
 }
 
-func (r *storageClientReconcile) EnsureClusterScopedResource(obj client.Object, extRes *providerpb.ExternalResource, useReplace bool) error {
-	obj.SetName(extRes.Name)
+func (r *storageClientReconcile) ensureResource(obj client.Object, rawObject []byte, useReplace bool, clusterScoped bool) error {
+	objectMeta := metav1.ObjectMeta{}
+	if err := json.Unmarshal(rawObject, &objectMeta); err != nil {
+		return err
+	}
+
+	obj.SetName(objectMeta.Name)
+	if !clusterScoped {
+		obj.SetNamespace(r.OperatorNamespace)
+	}
```

Comment:
This should not matter, even if you set a namespace on a cluster scooped resource the update/create should still work. There is no point in managing this manually
```suggestion
	obj.SetNamespace(r.OperatorNamespace)
```